src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,36 @@
-import React from 'react';
+import React, {
+  Dispatch,
+  SetStateAction,
+  useEffect,
+  useState,
+} from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+
+type Props = {
+  onSelect: Dispatch<SetStateAction<Todo | undefined>>;
+  todoSelected: Todo;
+};
+
+export const TodoModal: React.FC<Props> = ({ onSelect, todoSelected }) => {
+  const [todoLoading, setTodoLoading] = useState(true);
+  const [todoUser, setTodoUser] = useState<User>();
+
+  useEffect(() => {
+    getUser(todoSelected.userId)
+      .then((givenUser) => {
+        setTodoUser(givenUser);
+      })
+      .finally(() => setTodoLoading(false));
+  });
```

Comment:
```suggestion
  useEffect(() => {
    getUser(todoSelected.userId)
      .then((givenUser) => {
        setTodoUser(givenUser);
      })
      .finally(() => setTodoLoading(false));
  }, []);
```

let's add deps for `useEffect` because now `useEffect` will calling every time, please try to open modal and go to the network
look at requests
<img width="1348" alt="Снимок экрана 2023-10-31 в 11 08 29" src="https://github.com/mate-academy/react_dynamic-list-of-todos/assets/149401939/9381aa27-bc6a-4f83-8ef3-1536f36fe392">